internal/sudo/sudo.go

```diff
@@ -91,15 +93,18 @@ func NewSudoHandler(wasmKeeper *wasm.Keeper, moduleName string) Handler {
 
 func (s *Handler) SudoResponse(
 	ctx sdk.Context,
-	contractAddress sdk.AccAddress,
+	senderAddress sdk.AccAddress,
 	request channeltypes.Packet,
 	msg []byte,
 ) ([]byte, error) {
-	s.Logger(ctx).Debug("SudoResponse", "contractAddress", contractAddress, "request", request, "msg", msg)
-
-	if !s.wasmKeeper.HasContractInfo(ctx, contractAddress) {
-		s.Logger(ctx).Debug("SudoResponse: contract not found", "contractAddress", contractAddress)
-		return nil, fmt.Errorf("%s is not a contract address", contractAddress)
+	s.Logger(ctx).Debug("SudoResponse", "senderAddress", senderAddress, "request", request, "msg", msg)
+
+	if !s.wasmKeeper.HasContractInfo(ctx, senderAddress) {
+		s.Logger(ctx).Debug("SudoResponse: contract not found", "senderAddress", senderAddress)
+		if request.SourcePort == TransferPort {
```

Comment:
Please add some comment about these conditions.
It'll become a bit unclear for us after some time. Also it's not clear at all for newbies or third-party developers